### PR TITLE
Article Algorithms-Selection-Sort.md

### PR DESCRIPTION
**Pull request by varunu28**

### CODE REVIEW

**Comment by varunu28** [comment]
@alayek Let me know if we need to provide the implementation in any other programming language?
Also can we put up a video link of 5-7 minutes explaining the sort algorithm?

File: Algorithms-Selection-Sort.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Algorithm Selection Sort
+
+The selection sort algorithm sorts an array by repeatedly finding the minimum element (considering ascending order) from unsorted part and putting it at the beginning. The algorithm maintains two subarrays in a given array.
```
Comment (atjonathan) [inline Algorithms-Selection-Sort.md]: ~~from unsorted part~~ from **the** unsorted part

File: Algorithms-Selection-Sort.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+In every iteration of selection sort, the minimum element (considering ascending order) from the unsorted subarray is picked and moved to the sorted subarray.
+
+#### Example
+[Animation of SelectionSort](http://www.sorting-algorithms.com/selection-sort)
+
+```
+arr[] = 64 25 12 22 11
+
+#Placing the minimum element in arr[0...4] in the beginning
```
Comment (atjonathan) [inline Algorithms-Selection-Sort.md]: ~~#Placing~~ # Placing

File: Algorithms-Selection-Sort.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+11 25 12 22 64
+
+#Placing the minimum element in arr[1...4] in the beginning
+11 12 25 22 64
+
+#Placing the minimum element in arr[2...4] in the beginning
```
Comment (atjonathan) [inline Algorithms-Selection-Sort.md]: ~~#Placing~~ # Placing

File: Algorithms-Selection-Sort.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+#### Example
+[Animation of SelectionSort](http://www.sorting-algorithms.com/selection-sort)
+
+```
+arr[] = 64 25 12 22 11
+
+#Placing the minimum element in arr[0...4] in the beginning
+11 25 12 22 64
+
+#Placing the minimum element in arr[1...4] in the beginning
```
Comment (atjonathan) [inline Algorithms-Selection-Sort.md]: ~~#Placing~~ # Placing

File: Algorithms-Selection-Sort.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+11 12 25 22 64
+
+#Placing the minimum element in arr[2...4] in the beginning
+11 12 22 25 64
+
+#Placing the minimum element in arr[3...4] in the beginning
```
Comment (atjonathan) [inline Algorithms-Selection-Sort.md]: ~~#Placing~~ # Placing

**Comment by varunu28** [comment]
@atjonathan Implemented the changes. Please check

**Comment by atjonathan** [comment]
@koustuvsinha, @alayek what do you think?

**Comment by varunu28** [comment]
@alayek Please look into this PR and let me know if anything needs to be added

File: Algorithms-Selection-Sort.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
+# Algorithm Selection Sort
+
+The selection sort algorithm sorts an array by repeatedly finding the minimum element (considering ascending order) from the unsorted part and putting it at the beginning. The algorithm maintains two subarrays in a given array.  
+1) The subarray which is already sorted.
```
Comment (Rafase282) [inline Algorithms-Selection-Sort.md]: Please change to makrdown list.

1. 
2.

File: Algorithms-Selection-Sort.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
+# Algorithm Selection Sort
+
+The selection sort algorithm sorts an array by repeatedly finding the minimum element (considering ascending order) from the unsorted part and putting it at the beginning. The algorithm maintains two subarrays in a given array.  
+1) The subarray which is already sorted.  
+2) Remaining subarray which is unsorted.   
+
+In every iteration of selection sort, the minimum element (considering ascending order) from the unsorted subarray is picked and moved to the sorted subarray.
+
+#### Example
```
Comment (Rafase282) [inline Algorithms-Selection-Sort.md]:
```diff
-#### Example
+## Example
```

**Comment by crashuniverse** [comment]
@varunu28 PRs need to be squashed. Point 6 on https://github.com/FreeCodeCamp/FreeCodeCamp/blob/staging/CONTRIBUTING.md#contribution-guidelines.

I understand that github provides a way to merge the PR with squashed commits however it would be good to see contributors taking care of it too.

**Comment by Rafase282** [comment]
@varunu28 @crashuniverse this is our {squashing](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Git-Squash) guide.

File: Algorithms-Selection-Sort.md (hunk 7 above)
Comment (varunu28) [inline Algorithms-Selection-Sort.md]: @Rafase282 Done both the changes

**Comment by varunu28** [comment]
@Rafase282  Should I squash the commits?

**Comment by Rafase282** [comment]
Yes, it is always good to squash them. Many new contributors have problems with this part and since we acquired the ability to squash from here we have let up on asking people to do it a bit.

**Comment by atjonathan** [comment]
I'm going to squash these in the browser. Thanks for contributing :sparkles:

**Pull request merged**